http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoMetricsMonitorSpec.scala

```diff
@@ -0,0 +1,365 @@
+    ) shouldBe Some(1)
+  }
+
+  it should "health check apps" in {
+    val test =
+      leoMetricsMonitor.countAppsByHealth(List(cromwellAppAzure, galaxyAppGcp)).unsafeRunSync()(IORuntime.global)
+    // An up and a down metric for 5 services: wds, cbas, cbas-ui, cromwell galaxy
+    test.size shouldBe 10
+    List("wds", "cromwell", "cbas", "cbas-ui").foreach { s =>
+      test.get(
+        AppHealthMetric(CloudProvider.Azure,
+                        AppType.Cromwell,
+                        ServiceName(s),
+                        RuntimeUI.Terra,
+                        Some(azureContext),
+                        s != "cbas"
+        )
+      ) shouldBe Some(1)
+      test.get(
+        AppHealthMetric(CloudProvider.Azure,
+                        AppType.Cromwell,
+                        ServiceName(s),
+                        RuntimeUI.Terra,
+                        Some(azureContext),
+                        s == "cbas"
+        )
+      ) shouldBe Some(0)
+    }
+    test.get(
+      AppHealthMetric(CloudProvider.Gcp, AppType.Galaxy, ServiceName("galaxy"), RuntimeUI.Terra, None, true)
+    ) shouldBe Some(1)
+    test.get(
+      AppHealthMetric(CloudProvider.Gcp, AppType.Galaxy, ServiceName("galaxy"), RuntimeUI.Terra, None, false)
+    ) shouldBe Some(0)
+  }
+
+  it should "health check runtimes" in {
+    val test = leoMetricsMonitor.countRuntimesByHealth(List(jupyterAzure, rstudioGcp)).unsafeRunSync()(IORuntime.global)
+    // An up and a down for jupyter, rstudio, welder * 2
+    test.size shouldBe 8
+    // Jupyter Azure
+    List("Jupyter", "Welder").foreach { s =>
+      test.get(
+        RuntimeHealthMetric(CloudProvider.Azure,
+                            RuntimeImageType.stringToRuntimeImageType(s),
+                            RuntimeUI.Terra,
+                            Some(azureContext),
+                            true
+        )
+      ) shouldBe Some(1)
+      test.get(
+        RuntimeHealthMetric(CloudProvider.Azure,
+                            RuntimeImageType.stringToRuntimeImageType(s),
+                            RuntimeUI.Terra,
+                            Some(azureContext),
+                            false
+        )
+      ) shouldBe Some(0)
+    }
+    // RStudio GCP
+    List("RStudio", "Welder").foreach { s =>
+      test.get(
+        RuntimeHealthMetric(CloudProvider.Gcp,
+                            RuntimeImageType.stringToRuntimeImageType(s),
+                            RuntimeUI.Terra,
+                            None,
+                            s != "RStudio"
+        )
+      ) shouldBe Some(1)
+      test.get(
+        RuntimeHealthMetric(CloudProvider.Gcp,
+                            RuntimeImageType.stringToRuntimeImageType(s),
+                            RuntimeUI.Terra,
+                            None,
+                            s == "RStudio"
+        )
+      ) shouldBe Some(0)
+    }
+  }
+
+  // Data generators
```

Comment:
Was half thinking this test would be a good candidate for Scala [property based testing](https://scalacheck.org/) but I just hardcoded some seed data.

Reply:
Okay, I am not super familiar with this yet, but this feels like something that could potentially be reused elsewhere when testing and could therefore not be private?